Direction finding connectionless beacon sample app: add support for nRF5340

ppryga opened the pull request:
Update configuration to allow build of the sample application with nRF5340.
Add information about nRF5340 support in README.rst.

This PR requires following other PRs to be merged:
- ~~https://github.com/zephyrproject-rtos/zephyr/pull/36047 to be merged in upstream and cherry-picked into sdk-zephyr repository.~~
- ~~https://github.com/nrfconnect/sdk-nrf/pull/4763 to be merged into main.~~
- ~~Upmerge of sdk-zephyr repository has to be finished.~~

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-4769
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

mia-ko commented on samples/bluetooth/direction_finding_connectionless_tx/README.rst:
```diff
@@ -38,6 +38,14 @@ Configuration
 
 |config|
 
+This sample configuration is split into two parts:
+* generic configuration is available in :file:`prj.conf` file
+* board specific configuration is avaialbe in :file:`boards/<BOARD>.conf` file
```
```suggestion
This sample configuration is split into the following files:

* generic configuration is available in :file:`prj.conf` file
* board specific configuration is available in :file:`boards/<BOARD>.conf` file
```

ppryga replied:
Changed: added empty line and changed first sentence.

mia-ko commented on samples/bluetooth/direction_finding_connectionless_tx/README.rst:
```diff
@@ -38,6 +38,14 @@ Configuration
+* generic configuration is available in :file:`prj.conf` file
+* board specific configuration is avaialbe in :file:`boards/<BOARD>.conf` file
+
+Board specific configuration for the sample application is Bluetooth LE controller configuration.
+For :ref:`nRF5340 DK <ug_nrf5340>` Bluetooth LE controller is part of a ``child image`` aimed to run on network core.
+Configuration for a child image is stored in :file:`child_image/` subdirectory.
```
```suggestion
Board specific configuration involves configuring the Bluetooth LE controller.
For :ref:`nRF5340 DK <ug_nrf5340>`, the Bluetooth LE controller is part of a ``child image`` aimed to run on the network core.
Configuration for the child image is stored in :file:`child_image/` subdirectory.
```

ppryga replied:
Updated according to suggestion

mia-ko replied:
The first sentence here has one "configuration" too much, at the end of the sentence.

ppryga replied:
I'm sorry. It should be correct now.

mia-ko commented on samples/bluetooth/direction_finding_connectionless_tx/README.rst:
```diff
@@ -46,6 +54,8 @@ To build this sample with angle of arrival mode only, set ``OVERLAY_CONFIG`` to
 See :ref:`cmake_options` for instructions on how to add this option.
 For more information about using configuration overlay files, see :ref:`zephyr:important-build-vars` in the Zephyr documentation.
 
+To build this sample with angle of arrival mode only for :ref:`nRF5340 DK <ug_nrf5340>` add content of :file:`overlay-aoa.conf` to :file:`child_image/hci_rpmsg.conf`.
```
Make this the second sentence in the section
```suggestion
To build this sample for :ref:`nRF5340 DK <ug_nrf5340>`, with angle of arrival mode only, add content of :file:`overlay-aoa.conf` file to :file:`child_image/hci_rpmsg.conf` file.
```

ppryga replied:
Updated according to suggestion

mia-ko reviewed (changes requested):
It does not seem like the doc preview link included the changes that were made to the files... Strange!

mia-ko commented on samples/bluetooth/direction_finding_connectionless_tx/README.rst:
```diff
@@ -38,6 +38,15 @@ Configuration
+* generic configuration is available in :file:`prj.conf` file
+* board specific configuration is avaialbe in :file:`boards/<BOARD>.conf` file
+
+Board specific configuration involves configuring the Bluetooth LE controller configuration.
```
Something must have went wrong here... Seems to be one configuration too much, at the end of the sentence.

ppryga replied:
I'm sorry. It should be correct now.

ppryga commented:
@mia-ko Comments were adressed. Could you take another look?

mia-ko commented on samples/bluetooth/direction_finding_connectionless_tx/README.rst:
```diff
@@ -38,6 +38,15 @@ Configuration
+This sample configuration is split into the following two files:
+
+* generic configuration is available in :file:`prj.conf` file
+* board specific configuration is avaialbe in :file:`boards/<BOARD>.conf` file
```
```suggestion
* board specific configuration is available in :file:`boards/<BOARD>.conf` file
```

ppryga replied:
Corrected. thank you.

mia-ko reviewed (changes requested):
Just one typo, otherwise it looks good.

ppryga commented:
PR converted to Draft on request from @thst-nordic due to limited CI resources. Reviewing and approval process still continues without change.

ppryga commented:
Moved DTS overlay from `child_image/hci_rpmsg.overlay` to `child_image/hci_rpmsg/nrf5340dk_nrf5340_cpunet.overlay`.
Corrected antenna switches GPIO mapping in the DTS overlay.

ppryga commented:
@greg-fer could you take look on updated release notes?

greg-fer commented on doc/nrf/releases/release-notes-changelog.rst:
```diff
@@ -87,6 +87,7 @@ Bluetooth LE
 * Updated:
 
   * :ref:`ble_samples` - Changed the Bluetooth® sample Central DFU SMP name to :ref:`Central SMP Client <bluetooth_central_dfu_smp>`.
+  * :ref:`direction_finding_connectionless_rx` and :ref:`direction_finding_connectionless_tx` - Added default configuration for ``nrf52833dk_nrf52820``, ``nrf5340dk_nrf5340_cpuapp_and_cpuappns`` boards.
```
```suggestion
  * :ref:`direction_finding_connectionless_rx` and :ref:`direction_finding_connectionless_tx` samples - Added default configuration for ``nrf52833dk_nrf52820`` and ``nrf5340dk_nrf5340_cpuapp_and_cpuappns`` boards.
```

greg-fer replied:
@ppryga here you go.
